src/tsconfig/file_matcher.rs

```diff
@@ -0,0 +1,303 @@
+
+                if ends_with_glob {
+                    // Pattern already has wildcards, use as-is
+                    vec![pattern]
+                } else {
+                    // Non-glob pattern: expand to match directory
+                    // Strip trailing slash before adding /**
+                    let pattern_base = pattern.trim_end_matches('/');
+                    let mut patterns = Vec::new();
+
+                    // If looks like a file (has extension after last slash), also match exact
+                    if pattern
+                        .rsplit('/')
+                        .next()
+                        .is_some_and(|part| part.contains('.') && part != "." && part != "..")
+                    {
+                        patterns.push(pattern.clone());
+                    }
+
+                    patterns.push(format!("{pattern_base}/**"));
+                    patterns
+                }
+            })
+            .collect()
+    }
+
+    /// Test if a file matches this tsconfig's patterns.
+    ///
+    /// # Returns
+    ///
+    /// `true` if the file matches, `false` otherwise.
+    ///
+    /// # Algorithm
+    ///
+    /// 1. Normalize the file path (relative to tsconfig_dir with ./ prefix)
+    /// 2. Check files array first (highest priority, overrides exclude)
+    /// 3. Check if path matches any include pattern
+    /// 4. Check if path matches any exclude pattern
+    #[must_use]
+    pub fn matches(&self, file_path: &Path) -> bool {
+        // Normalize path for matching
+        #[allow(clippy::manual_let_else)] // Match is clearer here
+        #[cfg_attr(not(target_os = "windows"), allow(unused_mut))]
+        let mut normalized = match self.normalize_path(file_path) {
+            Some(p) => p,
+            None => return false, // Path can't be normalized
+        };
+
+        // On Windows, convert to lowercase for case-insensitive matching
+        #[cfg(target_os = "windows")]
+        {
+            normalized = normalized.to_lowercase();
+        }
+
+        // 1. Check files array first (absolute priority)
+        if let Some(files) = &self.files {
+            for file in files {
+                // Check both exact match and ends_with
+                if normalized == *file || normalized.ends_with(file) {
+                    return true; // Files overrides exclude
+                }
+            }
+            // If files specified but no match, continue to include/exclude
+            // (unless include is empty)
+            if self.include_patterns.is_empty() {
+                return false;
+            }
+        }
+
+        // 2. Check if empty patterns (match nothing case)
+        if self.include_patterns.is_empty() {
+            return false;
+        }
+
+        // 3. Test against include patterns
+        let mut included = false;
+        for pattern in &self.include_patterns {
+            if fast_glob::glob_match(pattern, &normalized) {
```

Comment:
tsconfig only supports `*` / `**` / `?` as a special char while fast-glob supports more. Not a big deal, but there are some bugs when `[` or `{` or `!` exists in a path.